libgenic/zeldovich.c

```diff
@@ -158,15 +159,20 @@ void displacement_fields(int Type) {
            &pstruct, NULL);
 
     /*This allocates the memory*/
-    pfft_complex * rho_k = petapm_get_rho_k();
+    pfft_complex * rho_k = petapm_alloc_rhok();
+
     gaussian_fill(petapm_get_fourier_region(),
 		  rho_k, All2.UnitaryAmplitude, All2.InvertPhase);
 
     petapm_force_c2r(rho_k, regions, functions);
+    myfree(rho_k);
+    myfree(regions);
     petapm_force_finish();
     double maxdisp = 0;
     int i;
-    for(i = 0; i < NumPart; i ++) {
+    #pragma omp parallel for reduction(max:maxdisp)
+    for(i = 0; i < NumPart; i++)
+    {
```

Comment:
Could you by the way make the `MPI_Reduce` a few lines lower in place? these one time variables are really eyesores.

Reply:
I'll pull forward another part of the cleanup from the scaledepgrowth branch.

Reply:
How do you like it now?